src/stores/uiStore.ts

```diff
@@ -167,6 +170,10 @@ class UiStore extends BaseStore {
     this.searchWalletName = Watchlists.All;
   };
 
+  setUserSelectedInstrument = (instrument: string) =>
```

Comment:
How about using accessors here? Testing getters & setters isn't smooth enough, but I find them more preferable in ts code.